resources/waterheater.rb

```diff
@@ -75,11 +75,14 @@ def self.apply_tank(model, runner, space, fuel_type, cap, vol, ef,
       new_heater.addToNode(storage_tank.supplyOutletModelObject.get.to_Node.get)
     end
 
+    if has_desuperheater
```

Comment:
If we just have a single argument, you can check `if not desuperheater_clg_coil.nil?` instead.